stdlib/public/core/CocoaArray.swift

```diff
@@ -47,23 +47,24 @@ internal struct _CocoaArrayWrapper : Collection {
   /// implementation of countByEnumeratingWith.
   func contiguousStorage(
     subRange: Range<Int>
-  ) -> UnsafeMutablePointer<AnyObject>
+  ) -> UnsafeMutablePointer<AnyObject>?
   {
     var enumerationState = _makeSwiftNSFastEnumerationState()
+    var dummyBuffer = [AnyObject]()
 
     // This function currently returns nil unless the first
     // subRange.endIndex items are stored contiguously.  This is an
     // acceptable conservative behavior, but could potentially be
     // optimized for other cases.
     let contiguousCount = withUnsafeMutablePointer(&enumerationState) {
-      self.buffer.countByEnumeratingWith($0, objects: nil, count: 0)
+      self.buffer.countByEnumeratingWith(
+        $0, objects: &dummyBuffer, count: 0)
```

Comment:
> there are a lot of Swift programs out there that already pass `nil` here

Evidence? Because that's pretty disconcerting. Foundation could certainly change their behavior for new programs. They're unlikely to, but we should not rely on that. "This has always worked in the past" is not a valid reason to violate documented nullability constraints.